src/libmain.c

```diff
@@ -765,16 +769,20 @@ static gint setup_config_dir(void)
 	return mkdir_result;
 }
 
-/* Signal handling removed since main_quit() uses functions that are
- * illegal in signal handlers
-static void signal_cb(gint sig)
+
+#ifdef G_OS_UNIX
+static gboolean signal_cb(gpointer user_data)
 {
+	gint sig = GPOINTER_TO_INT(user_data);
 	if (sig == SIGTERM)
```

Comment:
The idea was to make it like a normal signal handler, and since any future handlers are likely going to do something similar, or at least something small/simple, it seemed good to keep all the handling in one place.

At this point it would be cleaner not to pass the signal and just have it `sigterm_handler()` but I intend to submit a pull request "soon" to handle `SIGINT` so the config gets saved when using <kbd>Ctrl</kbd>+<kbd>C</kbd> from the terminal, in which case it's a mere matter of adding `|| sig == SIGINT` in the handler and another call to `g_unix_signal_add()` for `SIGINT`.

If you want, I can change it for now though, and then re-evaluate the code when there's more than one signal.